interfaces/builtin/desktop.go

```diff
@@ -38,7 +38,12 @@ const desktopBaseDeclarationSlots = `
   desktop:
     allow-installation:
       slot-snap-type:
+        - app
         - core
+    deny-connection:
+      on-classic: false
+    deny-auto-connection:
+      on-classic: false
```

Comment:
we can't express this atm but we probably would like to stop auto-connection if the slot is from an app. OTOH see the other point.

Reply:
It would be appropriate to switch to using a `slot-snap-type` constraint here once #13062 lands. Given that we've got the installation constraint working, would it be best to leave this out entirely for now, or wait for the other PR to land?

Reply:
in #13062 we don't have an installation constraint, this is probably fine here at least for now